browser/ui/webui/basic_ui.cc

```diff
@@ -50,65 +49,18 @@ content::WebUIDataSource* CreateBasicUIHTMLSource(
   return source;
 }
 
-// This is used to know proper timing for setting webui properties.
-// So far, content::WebUIController::RenderFrameCreated() is used.
-// However, it doesn't get called sometimes when reloading, or called when
-// RenderFrameHost is not prepared during renderer process is in changing.
-class BasicUI::BasicUIWebContentsObserver
-    : public content::WebContentsObserver {
- public:
-  BasicUIWebContentsObserver(BasicUI* host, content::WebContents* web_contents)
-      : WebContentsObserver(web_contents),
-        host_(host) {
-  }
-  ~BasicUIWebContentsObserver() override {}
-
-  // content::WebContentsObserver overrides:
-  void RenderViewReady() override {
-    host_->UpdateWebUIProperties();
-  }
-
- private:
-  BasicUI* host_;
-
-  DISALLOW_COPY_AND_ASSIGN(BasicUIWebContentsObserver);
-};
-
 BasicUI::BasicUI(content::WebUI* web_ui,
                  const std::string& name,
                  const GritResourceMap* resource_map,
                  size_t resource_map_size,
                  int html_resource_id,
                  bool disable_trusted_types_csp)
     : WebUIController(web_ui) {
-  observer_.reset(
-      new BasicUIWebContentsObserver(this, web_ui->GetWebContents()));
   Profile* profile = Profile::FromWebUI(web_ui);
   content::WebUIDataSource* source = CreateBasicUIHTMLSource(profile, name,
       resource_map, resource_map_size, html_resource_id,
       disable_trusted_types_csp);
   content::WebUIDataSource::Add(profile, source);
 }
 
-BasicUI::~BasicUI() {
```

Comment:
it appears to me we don't need this class at all

Reply:
Ok, I'll do in a separate issue. https://github.com/brave/brave-browser/issues/13955